### PR TITLE
doc: clarify bootstrap commands in hacking.rst

### PR DESCRIPTION
**Pull request by Alizter**

### CODE REVIEW

File: doc/hacking.rst [hunk 1]
```diff
@@ -26,12 +26,16 @@ invocation of ``ocamlopt`` or ``ocamlc`` via the ``bootstrap.ml`` OCaml script.
 and uses this binary to build everything else. As a convenience, ``dune.exe``
 at the root of the source tree executes this binary.
 
-``$ make dev`` takes care of bootstrapping if needed, but if you want to just
-run the bootstrapping step itself, build the ``dune.exe`` target with
+Running:
+```
+make dev
+```
+takes care of bootstrapping if needed, but if you want to just run the
```
Comment (snowleopard) [inline doc/hacking.rst:33]: What does "takes care of bootstrapping if needed" mean concretely?

Here is a (partially evaluated) guess:
* It builds `_boot/dune.exe`.
* It uses `_boot/dune.exe` to build "everything else" from the above paragraph. What does "everything else" include?

Comment (Alizter) [reply doc/hacking.rst:33]: We should really clarify that this is doing `./dune.exe build @install`. So everything else, means all the packages and importantly the dune binary.

File: doc/hacking.rst [hunk 2]
```diff
@@ -26,12 +26,16 @@ invocation of ``ocamlopt`` or ``ocamlc`` via the ``bootstrap.ml`` OCaml script.
...
+bootstrapping step itself, build the ``_boot/dune.exe`` target with
 
 .. code:: sh
 
-   make dev
+   make _boot/dune.exe
 
 Once you've bootstrapped dune, you should be using it to develop dune itself.
```
Comment (christinerose) [inline doc/hacking.rst:40]:
```suggestion
Once you've bootstrapped Dune, you should be using it to develop Dune itself.
```

**Review by christinerose — APPROVED** [review]
Just two capitalization corrections from dune to Dune.

File: doc/hacking.rst [hunk 3]
```diff
@@ -26,14 +26,20 @@ invocation of ``ocamlopt`` or ``ocamlc`` via the ``bootstrap.ml`` OCaml script.
 and uses this binary to build everything else. As a convenience, ``dune.exe``
 at the root of the source tree executes this binary.
 
-``$ make dev`` takes care of bootstrapping if needed, but if you want to just
-run the bootstrapping step itself, build the ``dune.exe`` target with
+Running:
+```
+make dev
+```
+bootstraps (if necessary) and does `./dune.exe build @install`.
```
Comment (snowleopard) [inline doc/hacking.rst:33]:
```suggestion
bootstraps (if necessary) and runs `./dune.exe build @install`.
```

**Review by snowleopard — APPROVED** [review]
Thanks!

**Pull request merged**